resolwe/flow/models/entity.py

```diff
@@ -58,27 +59,59 @@ class Meta(BaseCollection.Meta):
             ("owner_relation", "Is owner of the relation"),
         )
 
+    UNIT_HOUR = 'hr'
+    UNIT_CHOICES = (
+        (UNIT_HOUR, 'Hour'),
+    )
+
+    def validate_unique(self, exclude=None):
+        """Check unique constraints on the model.
+
+        Separate check for the combination of ``collection`` and ``category``
+        is required as ``category`` is optional and thus
+        ``Meta.unique_together`` cannot be used.
+        """
+        super().validate_unique(exclude)
+
+        duplicate_exists = Relation.objects.exclude(id=self.id).filter(
+            collection_c=self.collection, category=self.category, category__isnull=False
+        ).exists()
+
+        if duplicate_exists:
+            raise ValidationError("Combination of 'collection' and 'category' must be unique.")
+
     #: type of the relation
     type = models.ForeignKey('RelationType', on_delete=models.PROTECT)
```

Comment:
We could simply use a `CharField` with choices now. We do not need the `ordered` field of the `RelationType` anymore, since we have designed the `RelationPartition` with `label` and `position` fields.

Reply:
This should be a model, so different applications (e.g. resolwe-bio) can define different types of relations. You cannot predefine all of them here.

Reply:
True